awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -460,33 +458,48 @@ def delete_if_needed(self, existing_item, handle_response=True, on_delete=None):
                 else:
                     self.fail_json(msg="Unable to delete {0} {1}: {2}".format(item_type, item_name, response['status_code']))
         else:
-            if not handle_response:
-                return None
+            self.exit_json(**self.json_output)
+
+    def modify_associations(self, association_endpoint, new_association_list):
```

Comment:
This association method still isn't used anywhere that I can find in the current PR

I would love to get it in separately where we can make use of it in 1 new useful parameter and give it proper testing and attention.